app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/maps/GoogleMapsScreen.kt

```diff
@@ -1,49 +1,235 @@
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Autorenew
+import androidx.compose.material.icons.filled.MyLocation
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
+import com.github.leuludyha.domain.model.library.Mocks
+import com.github.leuludyha.domain.model.user.User
 import com.github.leuludyha.ibdb.R
+import com.github.leuludyha.ibdb.presentation.Orientation
+import com.github.leuludyha.ibdb.presentation.components.books.book_views.MiniBookView
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.model.CameraPosition
 import com.google.android.gms.maps.model.LatLng
 import com.google.maps.android.compose.*
+import kotlinx.coroutines.launch
 
 /**
- * For now, this screen only starts google maps and displays a few hardcoded markers at EPFL.
+ * This screen opens a google maps composable. Currently implemented features include user
+ * localisation and nearby users display. When clicking on a marker, it can open the actual
+ * google maps app to propose a route to get to the marker.
  */
 @Composable
 fun GoogleMapsScreen(
     navController: NavHostController,
-    initialLatitude: String?, //These are treated as strings because they are Double, but navController only accepts floats
-    initialLongitude: String?,
-    interestPoints: Array<String>?
+    paddingValues: PaddingValues,
+    viewModel: GoogleMapsScreenViewModel = hiltViewModel()
 ) {
-    //This below will be replaced by the current user's location later.
-    val epfl = LatLng(46.520536, 6.568318)
-
     val cameraPositionState = rememberCameraPositionState {
-        position = CameraPosition.fromLatLngZoom(epfl, 15f)
+        position = CameraPosition.fromLatLngZoom(viewModel.defaultLocation, 15f)
     }
 
+    val uiSettings by remember {
+        mutableStateOf(
+            MapUiSettings(
+                //I did my own implementation of this, I encountered problems and no documentation online
```

Comment:
What do you mean by 
>I did my own implementation of this, I encountered problems and no documentation online

?

Reply:
Maps compose, which is the implementation of google maps for jetpack compose, is still under development currently. Some features have very little documentation and tutorials from one year ago are not always consistent with how the library works currently. In the case of a button for locating the user, maps compose is supposed to have a default implementation that should be a one liner to enable (in the mapUIsettings). Unfortunately, I am unable to make it work (despite thoroughly trying). For that reason, I ended up implementing my own button for locating the user.

Reply:
Reformulated the comment to make it clearer. Tell me if it is clearer now please.